src/machine.cpp

```diff
@@ -6157,6 +6201,68 @@ GcCallSite* resolveDynamic(Thread* t, GcInvocation* invocation)
     ++i;
   }
 
+  object varArgsArray_ = makeObjectArray(t, numVarArgs);
+  PROTECT(t, varArgsArray_);
+
+  if (numVarArgs > 0) {
+    assertT(t, i < bootstrapArray->length());
+    GcArray *varArgsArray = reinterpret_cast<GcArray*>(varArgsArray_);
```

Comment:
Thanks. That clears things up. In this case though, varArgsArray is just a cast and varArgsArray_ is already protected. So I think we're good here?